nx/lib/nx/binary_backend.ex

```diff
@@ -1172,12 +1172,16 @@ defmodule Nx.BinaryBackend do
 
   @impl true
   def svd(
-        {u_holder, %{type: output_type} = s_holder, v_holder},
+        {%{shape: {m, _}} = u_holder, s_holder, %{shape: {_, n}} = v_holder} = outputs,
         %{type: input_type, shape: input_shape} = tensor,
         opts
       ) do
+    if m < n do
+      raise ArgumentError, "SVD not implemented for wide matrices (m x n 2-D tensors where m < n)"
```

Comment:
Just a nit: the `m x n` is kind of difficult to read, maybe:

```suggestion
      raise ArgumentError, "SVD not implemented for wide matrices (tensors with shape {m, n} where m < n)"
```

Reply:
Thanks! Brainfart, because this is how all of the other exceptions talk about shapes lol